src/main/java/com/github/manolo8/darkbot/gui/tree/editors/ImagePicker.java

```diff
@@ -0,0 +1,80 @@
+package com.github.manolo8.darkbot.gui.tree.editors;
+
+import com.github.manolo8.darkbot.config.ImageWrapper;
+import com.github.manolo8.darkbot.gui.AdvancedConfig;
+import eu.darkbot.api.config.ConfigSetting;
+import eu.darkbot.api.config.util.OptionEditor;
+import java.awt.Component;
+import java.awt.Dimension;
+import java.awt.HeadlessException;
+import java.io.File;
+import javax.swing.JButton;
+import javax.swing.JComponent;
+import javax.swing.JDialog;
+import javax.swing.JFileChooser;
+import javax.swing.filechooser.FileNameExtensionFilter;
+
+public class ImagePicker extends JButton implements OptionEditor<ImageWrapper> {
+    private JFileChooser fc;
+    private ImageWrapper image;
+
+    public ImagePicker() {
+        addActionListener(e -> {
+            if (fc == null) {
+                String path = image == null ? "." : image.getPath();
+                fc = new JFileChooser(new File(path)) {
+                    @Override
+                    protected JDialog createDialog(Component parent) throws HeadlessException {
+                        JDialog dialog = super.createDialog(parent);
+                        dialog.setAlwaysOnTop(true);
+                        return dialog;
+                    }
+                };
+                FileNameExtensionFilter filter = new FileNameExtensionFilter(
+                        "Images Files", "png", "jpeg", "jpg", "gif");
+                fc.setFileFilter(filter);
+                fc.setAcceptAllFileFilterUsed(false);
+            }
+
+            if (fc.showOpenDialog(null) == JFileChooser.APPROVE_OPTION) {
+                image.setPath(fc.getSelectedFile().getAbsolutePath());
+                setEditing(image);
+            }
+        });
+
+    }
+
+    @Override
+    public void setText(String text) {
+        int separatorIndex = text.lastIndexOf(File.separator, text.length() - 30);
```

Comment:
You need to search for indexOf, which returns the first occurrence (after len - 30). Using lastIndexOf you would always cut at the *last* separator, leaving just the filename

Also len - 30 can be negative, i don't think it's correct to call `String#indexOf(..., -10)`, so you need to ensure this is only called if string is above 30 chars (what you had before)

Reply:
there is two options i can use in the if statement after `text != null && text.length() > 30 &&`
`text.indexOf(File.separator, text.length() - 30) != -1`
or 
`text.substring(text.length() - 30).contains(File.separator)`

would either one of theses be a good idea?

Reply:
the implementation of contains is often implemented as indexOf != -1, at least in collections

I'd go for indexOf approach because contains requires you to substring, also, you can reuse the indexof result:
```
int sepIdx;
if (... && len > 30 && (sepIdx = text.indexOf(..., len - 30)) != -1) {
    substring(sepIdx, len)
}
```